pkg/dns-server/resolver/server.go

```diff
@@ -0,0 +1,211 @@
+package resolver
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"github.com/miekg/dns"
+
+	"github.com/Kong/kuma/pkg/core"
+)
+
+var (
+	simpleDNSLog = core.Log.WithName("dns-server-resolver")
+)
+
+type (
+	VIPList           map[string]string
+	SimpleDNSResolver struct {
+		sync.RWMutex
+		domain  string
+		address string
+		cidr    string
+		viplist VIPList
+		handler DNSHandler
+		ipam    IPAM
+	}
+)
+
+func NewSimpleDNSResolver(domain, ip, port, cidr string) (DNSResolver, error) {
```

Comment:
What will happen if KumaCP restarts? I guess we will allocate new virtual IPs for each service. Is it okay? Maybe we can somehow make `viplist` persistent?

Reply:
We had a discussion with @jakubdyszkiewicz about this. The basic conclusion here was that regenerating the new IPs should not be a problem for the services unless they have long-lived (resolve once) connections.
In any case this remark reminded me that I can send TTL for the response. Thanks.